### PR TITLE
fix: [CDS-53795]: fixed ami tags and filters to be array type at runtime

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger coverage

File: src/modules/10-common/components/MultiTypeTagSelector/MultiTypeArrayTagSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  tags: SelectOption[]
+  isLoadingTags?: boolean
+  name: string
+  errorMessage?: string
+  className?: string
+  tooltipProps?: DataTooltipInterface
+  expressions: string[]
+  initialTags?: object | string
+  label?: string
+  setAsArray?: boolean
```
Comment (pankajsingh-harness) [inline src/modules/10-common/components/MultiTypeTagSelector/MultiTypeArrayTagSelector.tsx:31]: We can remove this now

File: src/modules/10-common/components/MultiTypeTagSelector/MultiTypeTagSelector.tsx [hunk 2]
```diff
@@ -44,47 +28,12 @@ interface MultiTypeTagSelectorProps {
...
-    }
-    return tagError || getString('common.noAvailableTags')
-  }, [tagError, getString])
-
-  return (
-    <Text lineClamp={1} width={400}>
-      {getNoTagsComponent()}
-    </Text>
-  )
+  setAsArray?: boolean
```
Comment (pankajsingh-harness) [inline src/modules/10-common/components/MultiTypeTagSelector/MultiTypeTagSelector.tsx:31]: We can remove this now

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger coverage

**Review by pankajsingh-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**